### DISCUSSION TITLE
Covalent server won't start

### DISCUSSION THREAD

**Discussion by aacury** [AgnostiqHQ/covalent#1891] [Q&A]
Hi guys,

I'm having a bad time trying to run this [code](https://pennylane.ai/qml/demos/tutorial_univariate_qvr/). The problem is exactly at `os.system("covalent start")`.

FYI: I'm running the code on Ubuntu 20.04, Python 3.8.10, Covalent version 0.220.0

Also, when I asked for the log, it surprisingly showed "/bin/sh: 1: python: not found"

Thanks in advance

ps: I'm a total noob with Linux, so if my question is stupid, please be kind :P

**Comment by santoshkumarradha** [comment]
Hey @aacury, 

Could you try issuing the start command from a terminal where you had installed the covalent ? Just grab a terminal, `pip install covalent` and then once done in the same terminal issue `covalent start`. It appears that there might be an issue with the environment you’re running the code in compared to the environment where Covalent and Python are installed. We’ll close this issue for now, but feel free to reopen it if you encounter any further problems with starting Covalent.

**Reply by aacury** [reply]
Hi, thanks for your prompt reply. 

Well, still no luck. I did it just like you said, but after waiting for more than 5 minutes, the prompt remains idle (see the print screen
 attached). 

![Screenshot from 2023-12-20 14-07-07](https://github.com/AgnostiqHQ/covalent/assets/67163995/e05a9cad-6734-4d1a-b373-7f2416c924c7)

**Reply by santoshkumarradha** [reply]
Something seems to be messed up in installation/config. 

Could you try the following please ? 
clean all covalent related files and reinstall in fresh environment.

```bash
# Remove any existing installation of Covalent
pip uninstall covalent -y

# Clear the Covalent configuration and cache directories
rm -rf ~./config/covalent
rm -rf ~./cache/covalent

# Create a new Python virtual environment named 'covalent_env'
python3 -m venv covalent_env

# Activate the virtual environment
source covalent_env/bin/activate

# Update pip to its latest version within the virtual environment
pip install --upgrade pip

# Install the 'covalent' package from PyPI
pip install covalent

# Warning: The next command will delete all old files and database of Covalent
covalent purge -Hy

# Start covalent
covalent start

# Check the status of covalent
covalent status

# Note: To deactivate the virtual environment, you can use the 'deactivate' command
```

**Reply by santoshkumarradha** [reply]
Let me know if this works in. 

CC : @kessler-frost

**Reply by aacury** [reply]
It worked for the first time. However, after I stopped it and tried to start the server again, the prompt remained still. 

Do you recommend reinstalling Ubuntu (perhaps another release) and covalent all over?

**Reply by santoshkumarradha** [reply]
That's interesting, could you pass the OS details (which Ubuntu version) and also the python+covalent version so that we can see if we can reproduce it ? You should be able to do it by `cat /etc/lsb-release`. 

And while doing so, if you can also check the output of `covalent logs` and pass it here, that would be useful to further debug.

**Reply by aacury** [reply]
I'm running the code on Ubuntu 20.04, Python 3.8.10, Covalent version 0.220.0

A friend of mine helped me setting up a docker file as seen in [here](https://github.com/AgnostiqHQ/covalent/blob/develop/Dockerfile) and everything seems to be running smoothly for now. I'll keep you posted if anything bad happens.

By the way, I'd like to take this opportunity to ask you another question: looking at the _covalent.conf_ file, I noticed that it is possible to adjust some parameters such as _num_workers_ and _dask.threads_per_worker_. What are the optimal values for them? 

By "optimal", I mean that I intend to use as much processing power as I can in my computer (Intel i7 8550U 1.8Ghz, 16GB RAM). Can I set them to "auto", for example? 

(Also, if there is any other parameter to set to use more cors, for example, please let me know).

Thanks again

**Reply by santoshkumarradha** [reply]
I'm glad to hear that things are working out! It seems like the issue might be related to a configuration error. To answer your question, yes, you can find all the configurations you need [here](https://docs.covalent.xyz/docs/user-documentation/sd-docker), which should help you get started smoothly. By default, the Dask workers are processed-based and use the number of cores as the default number of workers. This should be a good setup for most cases, unless you have specific requirements.

If you want more control over the Dask cluster while using Docker, here's a suggestion: start the Covalent server in Docker without starting Dask, and then launch a Dask cluster outside the container on your local machine. This way, you can easily manage the workers and other settings. You can then use this external Dask cluster as an executor for your Covalent workflows by using the Dask executor and providing the scheduler address as shown [here](https://docs.covalent.xyz/docs/user-documentation/api-reference/executors/dask/).

However, the best approach depends on the specific purpose you're using Covalent for. For example, if you're using Covalent to connect to Slurm clusters, the Dask settings may not be as crucial.

Feel free to provide more details about your use case, and I can offer more tailored advice. Don't hesitate to ask if you have any further questions or need additional assistance!

**Comment by santoshkumarradha** [comment]
Closing this for now, but for further usage pattern, feel free to open a new dedicated discussion and we can continue there 😄
